.github/workflows/conda-build.yml

```diff
@@ -0,0 +1,64 @@
+name: conda-build
+on:
+  pull_request:
+    branches:
+      - main
+  push:
+    branches:
+      - main
+    paths:
+      - '.github/**'
+      - 'conda.recipe/**'
```

Comment:
```yml
  pull_request:
    branches:
      - main
```
this part means that the conda-build CI is executed on **every PR commit where the target branch is `main`** (so you misunderstood this part)

```yml
  push:
    branches:
      - main
```
this part means that the conda-build CI is executed on every push to `main`

```yml
  push:
    paths:
      - '.github/**'
      - 'conda.recipe/**'
```
this part means that the conda-build CI is executed on every commit where these files are changed, even when they are not on a PR to `main` or on `main`.

This is a relatively standard behavior because this basically ensures that it gets executed all of the time when needed.

Reply:
Thanks! I clearly misunderstood what the "branches" argument of `pull_request` was doing. I'm perfectly okay with this then.